sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -81,9 +89,9 @@ def fprime2(x):
 
     optimum = newton(func, x0=x0, fprime=fprime, fprime2=fprime2,
                      maxiter=70, tol=2e-8)
-    assert np.allclose(loss.inverse_link_function(optimum), y_true)
-    assert np.allclose(loss.pointwise_loss(y_true, optimum), 0)
-    assert np.allclose(get_gradients(y_true, optimum), 0, atol=1e-7)
+    assert_allclose(loss.inverse_link_function(optimum)[0], y_true)
+    assert_allclose(func(optimum)[0], 0, atol=1e-14)
```

Comment:
Just a question on superficial read: doesn't `func` return a scalar (I'm a bit confused by the indexing) and if it's doesn't (and returns a ndarray as `loss.pointwise_loss`) why does using newton minimizer with it works as an objective function ?

Maybe some light type annotations on `func` could have helped. mypy now runs as part of CI..

Reply:
🤔 No, `func` returns an array with one element. I tried to return scalars everywhere which fails badly because the methods of loss classes often call `reshape(-1)` internally. (And I have not yet completely figured out why, but this is another story).

I added a comment. I can also add type annotations. Can you provide an short example with ndarrays (of one element)?

Reply:
> Can you provide an short example with ndarrays (of one element)?

The typing for ndarray dimensions is not too standardized yet (https://github.com/numpy/numpy-stubs), but I think even,
```py
def func(x: np.ndarray) -> np.ndarray:
```
would have been useful as opposed to,
```py
def func(x: np.ndarray) -> float:
```

but anyway it's just a side comment,